components/camel-kafka/src/main/java/org/apache/camel/component/kafka/KafkaConsumer.java

```diff
@@ -460,7 +460,14 @@ public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
                     offset = -1L;
                 }
                 LOG.debug("Saving offset repository state {} from offsetKey {} with offset: {}", threadId, offsetKey, offset);
-                commitOffset(offsetRepository, partition, offset, true);
+                try {
+                    commitOffset(offsetRepository, partition, offset, true);
+                } catch (java.lang.Exception e) {
+                    LOG.error("Error saving offset repository state {} from offsetKey {} with offset: {}", threadId, offsetKey, offset);
+                    throw e;
+                } finally {
+                    lastProcessedOffset.clear();
```

Comment:
This finally block code is always exectued also for success. Is this really intended?

Reply:
Thank you for catching that. I'm going to decline the request and create a new one.

Reply:
Yes removal is required on success.  This fixes an issue where an exception on commit prevents the map entrry being removed.

Reply:
Thank you. 
@davsclaus Yes, doing it in the finally is correct.